keras/utils/data_utils.py

```diff
@@ -684,8 +695,18 @@ def get(self):
         """
         try:
             while self.is_running():
-                inputs = self.queue.get(block=True).get()
-                self.queue.task_done()
+                try:
+                    future = self.queue.get(block=True)
+                    inputs = future.get(timeout=30)
+                    self.queue.task_done()
+                except mp.TimeoutError:
+                    warnings.warn(
+                        "An input could not be retrieved."
+                        " It could be because a worker has died."
+                        "We do not have any information on the lost sample."
+                            .format(),
```

Comment:
I don't believe the format is useful.

Reply:
+1, maybe you intended to display the index in the warning?

Reply:
Yeah, but we cannot know the index in a generator.